2018/AsyncFromSyncIteratorContinuation.js

```diff
@@ -0,0 +1,37 @@
+'use strict';
+
+var GetIntrinsic = require('../GetIntrinsic.js');
+var callBound = require('../helpers/callBound.js');
+
+var Call = require('./Call.js');
+var CreateIterResultObject = require('./CreateIterResultObject.js');
+var IteratorComplete = require('./IteratorComplete.js');
+var IteratorValue = require('./IteratorValue.js');
+
+var PromiseResolve = require('./PromiseResolve.js');
+
+var $Promise = GetIntrinsic('%Promise%', true);
+var PerformPromiseThen = callBound('%Promise.prototype.then%', true);
+
+// https://www.ecma-international.org/ecma-262/10.0/#sec-asyncfromsynciteratorcontinuation
+
+module.exports = function AsyncFromSyncIteratorContinuation(result, promiseCapability) {
```

Comment:
the spec doesn't have the proper assertions here (see https://github.com/tc39/ecma262/pull/2050) but i think we should write a helper and the runtime check here.